source/FreeRTOS_IP.c

```diff
@@ -1817,8 +1817,20 @@ static eFrameProcessingResult_t prvProcessIPPacket( const IPPacket_t * pxIPPacke
 
                     case ipPROTOCOL_UDP:
                         /* The IP packet contained a UDP frame. */
-
-                        eReturn = prvProcessUDPPacket( pxNetworkBuffer );
+                        {
+                            const UDPPacket_t * pxUDPPacket = ( ( const UDPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
+                            uint16_t usLength;
+                            usLength = FreeRTOS_ntohs( pxUDPPacket->xUDPHeader.usLength );
+                            if( usLength > ( FreeRTOS_ntohs( pxIPHeader->usLength ) - uxIPHeaderSizePacket( pxNetworkBuffer ) ) )
```

Comment:
pxIPHeader can be obtained from NetworkBufferDescriptor_t pointer and we are passing the same to prvProcessUDPPacket.

Reply:
MISRA doesn't like the cast from `pucEthernetBuffer` to any of the structs. If a function needs such a cast, I tend to do so one time only.
For instance, `UDPPacket_t` already has a pointer to `IPHeader_t`, so you don't need the extra cast from `pucEthernetBuffer`.

It looks like this function should be compatible with IPv6, and that will need some adaptations.
The IPv6 header has `usPayloadLength`, which is defined as the "number of bytes after the IPv6 header".

So the patch works well for IPv4, it has an undefined behaviour for IPv6 packets.